### PR TITLE
ENH: add hausdorff distance method

### PR DESCRIPTION
**Pull request by kaushiksrini**
implements hausdorff distance, from #2010

### CODE REVIEW

File: geopandas/base.py [hunk 1]
```diff
@@ -2218,6 +2218,120 @@ def distance(self, other, align=True):
         """
         return _binary_op("distance", self, other, align)
 
+    def hausdorff_distance(self, other, align=True, densify=None):
+        """Returns a ``Series`` containing the Hausdorff distance to aligned `other`.
+        The Hausdorff distance is the largest distance consisting of any point in `self`
```
Comment (martinfleis) [inline geopandas/base.py:2223]:
```suggestion
        
        The Hausdorff distance is the largest distance consisting of any point in `self`
```

**Review by martinfleis — COMMENTED** [review]
Thanks! One minor note.

Can you also add a note to changelog?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Development version
 
+New methods:
+- Added ``hausdorff_distance()`` method from shapely to GeoSeries/GeoDataframe (#2909).
+
```
Comment (martinfleis) [inline CHANGELOG.md:7]:
```suggestion

- Added ``hausdorff_distance()`` method from shapely to GeoSeries/GeoDataframe (#2909).
```

**Review by martinfleis — APPROVED** [review]
Thanks! Just one small formatting suggestion.

**Comment by kaushiksrini** [comment]
it looks like a test is timing out and failing despite only changelog change. is there a way to rerun the checks?

File: doc/source/docs/reference/geoseries.rst [hunk 3]
```diff
@@ -31,6 +31,7 @@ General methods and attributes
    GeoSeries.y
    GeoSeries.z
    GeoSeries.get_coordinates
+   GeoSeries.hausdorff_distance
```
Comment (martinfleis) [inline doc/source/docs/reference/geoseries.rst:34]: Can you just move it right under `GeoSeries.distance` to have it a bit grouped by similarity? Thanks!

**Review by m-richards — APPROVED** [review]
Thanks! just two tidy suggestions in the docs.

**Comment by martinfleis** [comment]
Thanks @kaushiksrini!

**Pull request merged**